app/models/transcription.rb

```diff
@@ -30,7 +30,26 @@ def remove_files_from_storage
     export_files.map(&:purge)
   end
 
+  def lock(current_user)
+    update!(locked_by: current_user.login, lock_timeout: DateTime.now + 3.hours)
+  end
+
+  def locked?
+    lock_timeout && DateTime.now < lock_timeout
+  end
+
+  # Is transcription locked by a user that is not the current user?
+  def locked_by_different_user?(current_user)
+    locked? && current_user.login != locked_by
+  end
+
+  def fresh?(if_unmodified_since)
+    # defer to using datetime format used by Rails cache validation
+    DateTime.rfc2822(updated_at.rfc2822) == if_unmodified_since
```

Comment:
why do you have to case to rfc2822 twice to make this work? Doesn't it take a string rep and cast it to rfc2822?
`DateTime.rfc2822(updated_at)` or `DateTime.rfc2822(updated_at.to_s)`

Reply:
Ive updated the solution to mirror how Rails handles checking freshness. Here's a summary of what Ive done in tove:
1. Update "Last-Modified" header using `response.last_modified`, as done in Rails [`fresh_when` method](https://github.com/rails/rails/blob/v5.2.3/actionpack/lib/action_controller/metal/conditional_get.rb#L117). This will [transform date to Time.httpdate format](https://github.com/rails/rails/blob/v5.2.3/actionpack/lib/action_dispatch/http/cache.rb#L56).

2. Transform date incoming on the "If-Unmodified-Since" header to Time.rfc2822 format, [as done in Rails caching](https://github.com/rails/rails/blob/v5.2.3/actionpack/lib/action_dispatch/http/cache.rb#L12).
3. As done in Rails' `fresh?` method, check if resource has been updated since the If-Unmodified-Since date [with greater than or equal to comparison](https://github.com/rails/rails/blob/v5.2.3/actionpack/lib/action_dispatch/http/cache.rb#L25). Transform the `updated_at` date retrieved from the db to `httpdate` to imitate retrieving the date from a GET "Last-Modified" response header.